UnitTests/GitUI.Tests/Script/ScriptOptionsParserTests.cs

```diff
@@ -368,13 +368,130 @@ public void ParseScriptArguments_resolve_QuotedWithBackslashAtEnd()
         public void ParseScriptArguments_resolve_StringWithDoubleQuotes()
         {
             GitRevision gitRevision = new(ObjectId.Random());
-            gitRevision.Subject = "test string with \"double qoutes\" and escaped \\\"double qoutes\\\"";
+            gitRevision.Subject = "test string with \"double quotes\" and escaped \\\"double quotes\\\"";
 
             string result = ScriptOptionsParser.GetTestAccessor().ParseScriptArguments("{{sMessage}}", "sMessage", null, null, null, null, null, null, null, null, null, gitRevision, null, null, null, null, null, null);
-            result.Should().Be("\"test string with \\\"double qoutes\\\" and escaped \\\"double qoutes\\\"\"");
+            result.Should().Be("\"test string with \\\"double quotes\\\" and escaped \\\"double quotes\\\"\"");
 
             result = ScriptOptionsParser.GetTestAccessor().ParseScriptArguments("{sMessage}", "sMessage", null, null, null, null, null, null, null, null, null, gitRevision, null, null, null, null, null, null);
-            result.Should().Be("test string with \"double qoutes\" and escaped \\\"double qoutes\\\"");
+            result.Should().Be("test string with \"double quotes\" and escaped \\\"double quotes\\\"");
+        }
+
+        [Test]
+        public void Parse_should_parse_userInput()
+        {
+            _commands.GetService<ISimplePromptCreator>().Returns(new SimplePromptCreatorForTest(("input label1", "value", "foo1"), ("input label2", "value2", "bar1"), ("User input", string.Empty, "foo2")));
+            (string? arguments, bool abort) result = ScriptOptionsParser.Parse("{UserInput:input label1=value}_{{UserInput:input label2=value2}}_{UserInput}_{{UserInput}}",
+                _commands, owner: null, _scriptHostControl);
+
+            result.arguments.Should().Be(@"foo1_""bar1""_foo2_""foo2""");
+            result.abort.Should().Be(false);
+        }
+
+        [Test]
+        public void Parse_should_parse_userInput_without_default_value()
+        {
+            _commands.GetService<ISimplePromptCreator>().Returns(new SimplePromptCreatorForTest(("input label1", string.Empty, "foo1")));
+            (string? arguments, bool abort) result = ScriptOptionsParser.Parse("{UserInput:input label1}",
+                _commands, owner: null, _scriptHostControl);
+
+            result.arguments.Should().Be(@"foo1");
+            result.abort.Should().Be(false);
+        }
+
+        [Test]
+        public void Parse_should_parse_userInput_and_replace_all_with_same_label()
+        {
+            _commands.GetService<ISimplePromptCreator>().Returns(new SimplePromptCreatorForTest(("input label1", "value", "foo1")));
```

Comment:
```suggestion
            _commands.GetService<ISimplePromptCreator>().ReturnsOnce(new SimplePromptCreatorForTest(("input label1", "value", "foo1")));
```
in order to verify that the dialog pops up only once - if NSubstitute supports it.

Reply:
Not possible but checked by line 453 (where we check that the input mock is displayed only the number of timed configured).